manager/integration/tests/test_system_backup_restore.py

```diff
@@ -100,3 +106,118 @@ def test_system_backup_and_restore_volume_with_data(client, volume_name, set_ran
     restored_volume = wait_for_volume_healthy(client, volume_name)
 
     check_volume_data(restored_volume, data)
+
+
+@pytest.mark.system_backup_restore   # NOQA
+def test_system_backup_with_volume_backup_policy_if_not_present(client, volume_name, set_random_backupstore):  # NOQA
+    """
+    Scenario: system backup with volume backup policy (if-not-present) should
+              only create volume backup when there is no existing backup in
+              the volume.
+
+    Issue: https://github.com/longhorn/longhorn/issues/5011
+
+    Given a volume is created.
+
+    When system backup (system-backup-1) has no volume backup policy.
+    And system backup (system-backup-1) created.
+    Then system backup has volume backup policy (if-not-present).
+    And system backup is in state (Ready).
+    And volume has backup count (1).
+
+    When system backup (system-backup-2) has volume backup policy
+         (if-not-present).
+    And system backup (system-backup-2) created.
+    Then system backup is in state (Ready).
+    And volume has backup count (1).
```

Comment:
Can we extend the test case to verify  - if the existing backup doesn't have the latest data, the system backup should trigger a volumes backup?

Next steps:
When Data is written in the volume (No backup taken for the volume)
AND system backup (system-backup-3) created.
Then system backup is in state (Ready).
And volume has backup count (2).

Reply:
The if-not-present policy only checks if the volume has an existing backup, it doesn't check if the latest backup reflects the most recent data.

Reply:
oh okay, do you think we should file a ticket to improve this one in future?
cc: @innobead

Reply:
We should update the doc as well to explain it well if needed.  Sounds good to me to create a ticket for the improvement later.

Reply:
https://github.com/longhorn/longhorn/issues/6027